vllm/entrypoints/openai/run_batch.py

```diff
@@ -100,6 +101,28 @@ async def run_request(chat_serving: OpenAIServingChat,
 
     return batch_output
 
+MAX_CONCURRENT_REQUESTS = 10000
+
+async def process_requests(input_file: str, openai_serving_chat):
+    sem = Semaphore(MAX_CONCURRENT_REQUESTS)
+    
+    async def run_request_with_semaphore(request):
+        async with sem:
+            
+            return await run_request(openai_serving_chat, request)
+    
+    response_futures: List[Awaitable[BatchRequestOutput]] = []
+    
+    # Read file contents asynchronously
+    file_contents = (await read_file(input_file)).strip().split("\n")
+    total_lines = len(file_contents)
+    
```

Comment:
Let's figure out whether the engine queuing should be fixed instead, but if we decide to limit the requests we submit, I wonder if we should just not create the asyncio requests instead.

Something like

```
finished_requests = []
response_futures: List[Awaitable[BatchRequestOutput]] = []

for request_json in tqdm(file_contents, total=total_lines, desc="Running requests"):
        request = BatchRequestInput.model_validate_json(request_json.strip())
        response_futures.append(run_request_with_semaphore(request))
        if len(response_futures) >= MAX_CONCURRENT_REQUESTS:
            recently_finished, response_futures = await asyncio.wait(response_futures)
            finished_requests.extend(recently_finished)

finished_requests.extend(await asyncio.gather(*response_futures))
```

This way we can more accurately track the progress of the job.